src/core/thread/mle.cpp

```diff
@@ -788,15 +788,20 @@ ThreadError Mle::AppendLeaderData(Message &aMessage)
     return aMessage.Append(&mLeaderData, sizeof(mLeaderData));
 }
 
+void Mle::FillNetworkDataTlv(NetworkDataTlv &aTlv, bool aStableOnly)
```

Comment:
AppendNetworkData() appends binary data to the message. And the TLV object it uses is a temporary one. I also check that there is no public method to access internal buffer of a Message. Do you mean I should Read() data written by AppendNetworkData(), then modify the data by SetType(), finally rewrite data to the Message?